### DISCUSSION TITLE
Using The Phoebe Legacy Backend?

### DISCUSSION THREAD

**Discussion by dmdixon** [phoebe-project/phoebe1#24] [Q&A]
Although I somehow managed to do it before I am currently having a lot of trouble installing the Phoebe Legacy backend as an importable python module. Originally I was able to download the [stripped version](https://github.com/phoebe-project/phoebe1/tree/phoebe_legacy) and easily install it with the following.

python setup_static.py build
python setup_static.py install

Everything still does appear to compile, and the library is added to my anaconda distribution, but when I try to import phoebe_legacy I get an undefined symbol error. From what I read online this usually means that something went wrong during the compilation step. I can see that distutils is deprecated and so I think maybe that is the reason I could install some years back with an earlier version of python but not now.

Because I am not too familiar with compiling c programs I wanted to ask what do I need to change in the setup_static.py file to make sure things compiled properly?

gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Python 3.9.13

**Comment by aprsa** [comment]
Let's try this the old-fashioned way first: can you pull the actual legacy sources and build the library and the wrapper from scratch?
```
git clone https://github.com/phoebe-project/phoebe1 legacy
cd legacy/phoebe-lib
./configure
make
sudo make install
```
Provided that there are no errors, this should leave you with the legacy phoebe library installed system-wide.

Next, you should install the wrapper in your virtual environment. Activate it and issue:
```
cd ../phoebe-py
python setup.py build
python setup.py install
```
Again, assuming no errors, this should leave you with a working wrapper. To test:
```
python
>>> import phoebeBackend as legacy
>>> legacy.init()
0
```
If you run into any issues, paste the error here and we'll troubleshoot further.

**Reply by dmdixon** [reply]
I now believed I have successfully installed phoebe legacy. It looks like by default it was saved in the /user/local/lib/ directory. The top level of the directory looks like this.

_"cmake  libphoebe-1.0.so  libphoebe.a  libphoebe.la  libphoebe.so  libwd-2007-08-15.so  libwd.a  libwd.la  libwd.so  python3.8"_

 Now I seem to be running into an issue at the **python setup.py build** step. The error I get is the following.

_"/home/don/anaconda3/compiler_compat/ld: cannot find -lphoebe: No such file or directory"_

It seemed to me that the installed library was not in the search path so I included /usr/local/lib/ to my LD_LIBRARY_PATH environment variable and I added it to the package_dir argument of setuptools.setup, but I still get the same error.

Any idea what I am missing here? Thank you very much for the help!

**Comment by aprsa** [comment]
Someone with (more? any?) anaconda experience might help better here; I never used anaconda, I only ever use venv, so I'm really not sure how anaconda links a system library, or how an external library (such as phoebe) can be installed into the anaconda system. But I would try to figure out one of those two; if all else failed, I'd try to symlink a system-wide phoebe library to wherever anaconda expects to find its libraries. That might cause compiler/build system version conflicts but it also might work.

**Reply by aprsa** [reply]
Shooting in the dark here: perhaps compiling phoebe-lib the same way as above, but within anaconda environment and without sudo, might do the trick.

**Reply by dmdixon** [reply]
After reading up a bit on setuptools/distutils it turns out all I really needed to do was add search path options for the c library/header files. 

Since by default phoebe-lib is saved to the /usr/local directory, setting library_dirs = ['/usr/local/lib'] and include_dirs=[numpy.get_include(),'/usr/local/lib'] in the distutils.core Extension class instance before running python setup.py build fixed the issue.

I am now able to import legacy and run it as a backend compute option in the current version of Phoebe.

Thank you for your help!

**Reply by aprsa** [reply]
Great! Just note that the stripped down version you refer to above is out of date. So you should use the library you pulled from github and then tweak phoebe-py's [dist,setup]utils to point to the system-wide installation.
